pkg/templating/api.go

```diff
@@ -17,25 +17,39 @@ limitations under the License.
 package templating
 
 import (
+	"context"
+	"strconv"
 	"strings"
 
-	"github.com/crossplane/templating-controller/pkg/resource"
-
+	"github.com/pkg/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
 	"github.com/crossplane/crossplane-runtime/pkg/meta"
-
 	"github.com/crossplane/crossplane/pkg/stacks"
+
+	"github.com/crossplane/templating-controller/pkg/resource"
 )
 
-// Constants used for determining whether the defaulting annotations should be
-// removed or not.
+// Error strings.
 const (
-	RemoveDefaultAnnotationsKey       = "templatestacks.crossplane.io/remove-defaulting-annotations"
-	RemoveDefaultAnnotationsTrueValue = "true"
+	errDeleteChildResource = "cannot delete child resource"
+	errPriorityToInt       = "cannot convert deletion priority into integer"
 )
 
+// Constants used for annotations.
+const (
+	RemoveDefaultAnnotationsKey         = "templatestacks.crossplane.io/remove-defaulting-annotations"
+	RemoveDefaultAnnotationsTrueValue   = "true"
+	DeletionPriorityAnnotationKey       = "templatestacks.crossplane.io/deletion-priority"
+	DeletionPriorityAnnotationZeroValue = "0"
+)
+
+const minInt = -(int(^uint(0) >> 1)) - 1
```

Comment:
I feel like this dark magic could use a comment. It seems like it's the same as `math.MinInt64` on the Go playground - I'm guessing this is working around the fact that `strconv.Atoi` returns an `int`, which is ["either 32 or 64 bits"](https://golang.org/ref/spec#Numeric_types)? Perhaps we should use an `int64` and `strconv.ParseInt("0", 10, 8)` to bound this at 256 priority levels?